kuma/javascript/src/header/main-menu.jsx

```diff
@@ -0,0 +1,255 @@
+                            'https://github.com/mdn/sprints/issues/new?template=issue-template.md&projects=mdn/sprints/2&labels=user-report&title={{PATH}}'
+                    },
+                    {
+                        label: gettext('Report a bug'),
+                        external: true,
+                        url: 'https://bugzilla.mozilla.org/form.mdn'
+                    }
+                ]
+            }
+        ],
+        [locale]
+    );
+
+    /**
+     * Handles all interaction events
+     * @param {Object} event - The Event object
+     */
+    function interactionHandler(event: Event) {
+        let mediaQuery = window.matchMedia('(min-width: 47.9375em)');
+        let currentTarget = event.target;
+
+        /* currentTarget instanceof 'HTMLButtonElement' is added
+           to keep Flow happy: https://github.com/facebook/flow/issues/218#issuecomment-74119319 */
+        if (
+            mediaQuery.matches &&
+            (event.type === 'mouseover' || event.type === 'mouseout') &&
+            currentTarget instanceof HTMLButtonElement &&
+            currentTarget.classList.contains('top-level-entry')
+        ) {
+            if (event.type === 'mouseover') {
+                if (currentTarget.nextElementSibling) {
+                    currentTarget.nextElementSibling.setAttribute(
+                        'aria-hidden',
+                        'false'
+                    );
+                }
+            } else if (event.type === 'mouseout') {
+                if (currentTarget.nextElementSibling) {
+                    currentTarget.nextElementSibling.setAttribute(
+                        'aria-hidden',
+                        'true'
+                    );
+                }
+            }
+        } else if (
+            event.type === 'touchstart' &&
+            currentTarget instanceof HTMLButtonElement &&
+            currentTarget.classList.contains('top-level-entry')
+        ) {
+            let subMenu = currentTarget.nextElementSibling;
+
+            if (subMenu) {
+                let currentAriaState = subMenu.getAttribute('aria-hidden');
+                let newAriaState =
+                    currentAriaState === 'true' ? 'false' : 'true';
+                subMenu.setAttribute('aria-hidden', newAriaState);
+                subMenu.classList.toggle('show');
+            }
+        }
+    }
+
+    /**
+     * Hide the visible submenu in main navigation
+     */
+    function hideVisibleSubMenu() {
+        let visibleSubMenu = document.querySelector(
+            'li ul[aria-hidden="false"]'
+        );
+        if (visibleSubMenu) {
+            visibleSubMenu.setAttribute('aria-hidden', 'true');
+            visibleSubMenu.classList.toggle('show');
+        }
+    }
+
+    useEffect(() => {
+        const mediaQuery = window.matchMedia('(min-width: 47.9375em)');
+
+        if (!mediaQuery.matches) {
+            hideVisibleSubMenu();
+        }
+    }, [mdnDocument]);
```

Comment:
@peterbe This is the only portion I still have in the `useEffect`. I believe you have another option for how we can ensure the submenu closes upon navigation?